### ISSUE TITLE
Is it possible to return a Stream (boxed and pinned, I guess) from a Resolver?

### ISSUE THREAD

**Issue by Zizico2** [async-graphql/async-graphql#1369]
It would seem natural for async-graphql to support this since the data source being streamable happens regularly. And would this have a positive performance impact, I assume. I'm collecting a `Stream` to a `Vec` in the resolver, and then `async-graphql itself is iterating through that `Vec` again (or is it?).

Could this be possible with Iterators as well?

**Comment by sunli829** [comment]
I assume you mean Subscription?

https://async-graphql.github.io/async-graphql/en/subscription.html

**Comment by Zizico2** [comment]
No. I mean a list. I think GraphQL lists should be represented as `iterators` and/or `streams`. `async-graphql` is already iterating through the data that is being returned. I should be able to perform transforms on that data lazily, through `.map` or `.filter`. In other words, I shouldn't have to `.collect` before returning stuff in a resolver, as `async-graphql` is going to iterate through all the data anyway. Instead of me looping through the data, and then `async-graphql` looping through the data again, it should be just one loop, if I want to. This should be noticeable in big lists, where iterations take a *relatively* long time.
